core/string/ustring.cpp

```diff
@@ -4664,6 +4664,75 @@ String String::sprintf(const Array &values, bool *error) const {
 					in_format = false;
 					break;
 				}
+				case 'v': { // Vector2/3/4/2i/3i/4i
+					if (value_index >= values.size()) {
+						return "not enough arguments for format string";
+					}
+
+					Array numberValues = Array();
+
+					// Get values from the correct type of vector
+					switch (values[value_index].get_type()) {
+						case Variant::VECTOR2:
+						case Variant::VECTOR2I: {
+							Vector2 value = values[value_index];
+							numberValues.push_back(value.x);
+							numberValues.push_back(value.y);
+
+							break;
+						}
+						case Variant::VECTOR3:
+						case Variant::VECTOR3I: {
+							Vector3 value = values[value_index];
+							numberValues.push_back(value.x);
+							numberValues.push_back(value.y);
+							numberValues.push_back(value.z);
+
+							break;
+						}
+						case Variant::VECTOR4:
+						case Variant::VECTOR4I: {
+							Vector4 value = values[value_index];
+							numberValues.push_back(value.x);
+							numberValues.push_back(value.y);
+							numberValues.push_back(value.z);
+							numberValues.push_back(value.w);
+
+							break;
+						}
+						default: {
+							return "a vector is required";
+						}
+					}
+
+					String str = "(";
+					for (int i = 0; i < numberValues.size(); i++) {
+						double val = numberValues[i];
```

Comment:
Creating an `Array` seems overkill, we can do it in a simpler way:

```suggestion
					int count;
					// Get values from the correct type of vector
					switch (values[value_index].get_type()) {
						case Variant::VECTOR2:
						case Variant::VECTOR2I: {
							count = 2;
						} break;
						case Variant::VECTOR3:
						case Variant::VECTOR3I: {
							count = 3;
						} break;
						case Variant::VECTOR4:
						case Variant::VECTOR4I: {
							count = 4;
						} break;
						default: {
							return "%v requires a vector type (Vector2/3/4/2i/3i/4i)";
						}
					}
					Vector4 vec = values[value_index];
					String str = "(";
					for (int i = 0; i < count; i++) {
						real_t val = vec[i];
```

Reply:
That's much smarter! Thanks a lot! I didn't know that you could parse any vector type as a Vector4.